evergreen.yml

```diff
@@ -286,6 +286,30 @@ functions:
       params:
         file: mongosql-odbc-driver/expansions.yml
 
+  "compile release with debug info and asan flag":
+    - command: shell.exec
+      params:
+        shell: bash
+        working_dir: mongosql-odbc-driver
+        script: |
+          ${prepare_shell}
+          export RUSTFLAGS="-Z sanitizer=leak"
```

Comment:
Did you try something else than "leak", we use "address" on unit tests and it seems that it would useful too.
From the [doc](https://doc.rust-lang.org/beta/unstable-book/compiler-flags/sanitizer.html), "leak" is already on by default on Linux.

Reply:
I tried with `address` as we have in the unit tests, and with `memory`.  
They both fail, "address" causes an undefined symbol error for one of the crates that we import `serde_derive`, although it isn't imported for the unit tests which is why our unit tests don't hit the same issue. 
```
undefined symbol: __asan_option_detect_stack_use_after_return
```
The `memory` flag fails in another crate called `futures-task`:
   Compiling futures-task v0.3.28

Adding the `sanitizer=leak` flag did cause the integration tests to fail with the `ERROR: LeakSanitizer: detected memory leaks` errors which we don't see without the flag set so I would think it is testing something different than what is default.  

I updated the description to say that after fixing the integration tests, I was not able to get a test failure from my introduced memory leak.  I incorrectly thought the error was due to the changes in the library but it was in the tests themselves.  I confirmed that the code to add the leak is run by adding logging, it's not getting caught unfortunately.

Reply:
> I was not able to get a test failure from my introduced memory leak.

Are you completely sure that you were actually causing a memory leak? I just want to make sure because that seems kind of odd haha. Also, in the description you wrote, `After fixing the memory leaks in the integration tests, I was not able to capture the memory leak in the library.` So how do we know there is a memory leak in the library if we can't capture it?

Reply:
Yes, if I put the same leak code in the integration test code (integration_test/src/test_runner/mod.rs) or the connection test it will detect the memory leak.  The `Box::leak` method will take ownership of a Box and prevents it from being dropped.  I added that in a few functions in `odbc/src/api/functions.rs` and added logging before and after to confirm the leak code was being executed. 

>So how do we know there is a memory leak in the library if we can't capture it?
The integration tests have memory leaks that are fixed by this PR.  I injected the memory leak with `Box::leak` in the driver/library code and was running the test with the `RUSTFLAGS="-Z sanitizer=leak"` flag expecting for it to detect the memory leak but it didn't.  So we don't know if there is a memory leak in the existing driver code but we also can't detect it through the integration test.  The leak tester does run with the unit tests so that's how we will confirm there are no leaks.